mypy/checkexpr.py

```diff
@@ -227,9 +229,10 @@ def analyze_ref_expr(self, e: RefExpr, lvalue: bool = False) -> Type:
 
     def analyze_var_ref(self, var: Var, context: Context) -> Type:
         if var.type:
-            if isinstance(var.type, Instance):
-                if self.is_literal_context() and var.type.last_known_value is not None:
-                    return var.type.last_known_value
+            var_type = get_proper_type(var.type)
+            if isinstance(var_type, Instance):
+                if self.is_literal_context() and var_type.last_known_value is not None:
+                    return var_type.last_known_value
                 if var.name() in {'True', 'False'}:
                     return self.infer_literal_expr_type(var.name() == 'True', 'builtins.bool')
             return var.type
```

Comment:
Do things end up breaking if we return `var_type` and make the return value be ProperType?

Reply:
No, but again I would like to keep the original type for error message (if we will decide this at some point, I think there is even an issue about this).